WearComplicationDataSourcesTestSuite/Wearable/src/main/java/com/example/android/wearable/wear/wearcomplicationproviderstestsuite/WeightedElementsDataSourceService.kt

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright 2021 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.example.android.wearable.wear.wearcomplicationproviderstestsuite
+
+import android.app.PendingIntent
+import android.content.ComponentName
+import android.graphics.Color
+import android.graphics.drawable.Icon
+import android.os.Build
+import androidx.annotation.RequiresApi
+import androidx.datastore.core.DataStore
+import androidx.wear.watchface.complications.data.ComplicationData
+import androidx.wear.watchface.complications.data.ComplicationText
+import androidx.wear.watchface.complications.data.ComplicationType
+import androidx.wear.watchface.complications.data.MonochromaticImage
+import androidx.wear.watchface.complications.data.PlainComplicationText
+import androidx.wear.watchface.complications.data.WeightedElementsComplicationData
+import androidx.wear.watchface.complications.datasource.ComplicationDataSourceService
+import androidx.wear.watchface.complications.datasource.ComplicationRequest
+import androidx.wear.watchface.complications.datasource.SuspendingComplicationDataSourceService
+import kotlin.random.Random
+
+/**
+ * A complication provider that supports only [ComplicationType.WEIGHTED_ELEMENTS] and cycles
+ * through the possible configurations on tap. The value is randomised on each update.
+ *
+ * Note: This subclasses [SuspendingComplicationDataSourceService] instead of [ComplicationDataSourceService] to support
+ * coroutines, so data operations (specifically, calls to [DataStore]) can be supported directly in the
+ * [onComplicationRequest].
+ *
+ * If you don't perform any suspending operations to update your complications, you can subclass
+ * [ComplicationDataSourceService] and override [onComplicationRequest] directly.
+ * (see [NoDataDataSourceService] for an example)
+ */
+@RequiresApi(Build.VERSION_CODES.TIRAMISU)
+class WeightedElementsDataSourceService : SuspendingComplicationDataSourceService() {
+
+    override suspend fun onComplicationRequest(request: ComplicationRequest): ComplicationData? {
+        if (request.complicationType != ComplicationType.WEIGHTED_ELEMENTS) {
```

Comment:
Is null the right result here? or NoDataComplicationData?

https://cs.android.com/androidx/platform/frameworks/support/+/androidx-main:wear/watchface/watchface-complications-data-source-ktx/src/main/java/androidx/wear/watchface/complications/datasource/SuspendingComplicationDataSourceService.kt;l=45?q=SuspendingComplicationDataSourceService

Reply:
`NoDataComplicationData` is better, you're right (confirmed with Alex)